### PR TITLE
Split diff and report job to allow diff without the need of credentials

### PR DESCRIPTION
**Pull request by raffaele-oplabs**
This pull request includes several changes to CircleCI configuration files to simplify workflows and remove unnecessary parameters. The most important changes include the removal of the `is_pull_request` parameter, renaming workflows for clarity, and restructuring jobs to separate diff computation and reporting.

Simplification of CircleCI configuration:

* [`.circleci/artifact_upload_config.yml`](diffhunk://#diff-a0011e6b01ec3cc8297076866ff565f2a541b62752ad1f72f0ab03c85c29ca3fL13-L15): Removed `is_pull_request` parameter and renamed the `upload-chain-artifacts` workflow to `upload-chain-artifacts-workflow`. [[1]](diffhunk://#diff-a0011e6b01ec3cc8297076866ff565f2a541b62752ad1f72f0ab03c85c29ca3fL13-L15) [[2]](diffhunk://#diff-a0011e6b01ec3cc8297076866ff565f2a541b62752ad1f72f0ab03c85c29ca3fL105-R102)
* [`.circleci/config.yml`](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L39-R39): Removed the logic for setting `is_pull_request` and simplified the generation of `parameters.json`.
* [`.circleci/main_config.yml`](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L13-L15): Removed `is_pull_request` parameter and restructured jobs to separate diff computation and reporting. Added new parameters `save_to` and `read_from` for handling diff results. [[1]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L13-L15) [[2]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L256-R253) [[3]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L266-R274) [[4]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R284) [[5]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L311-R335) [[6]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619R369) [[7]](diffhunk://#diff-a7236285ae969f09dbccee34ec0fa1ea2646f33fed37fb1d4ba5e03e66510619L401-R432)

### CODE REVIEW

**Comment by opgitgovernance** [comment]
Detected nonzero diff in genesis. This PR modifies the genesis data of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/49095).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here](https://circleci.com/gh/ethereum-optimism/superchain-registry/49094).

**Comment by opgitgovernance** [comment]
Detected nonzero diff in rollup config. This PR modifies the configuration of at least one chain in the registry, which may or may not have been intended. You can see the detail [here]().

File: .circleci/main_config.yml [hunk 1]
```diff
@@ -263,32 +260,18 @@ jobs:
...
-            BASE_BRANCH=$(jq -r '.base.ref' api_response.json)
-
-            echo "HEAD_SHA=$HEAD_SHA"
-            echo "BASE_BRANCH=$BASE_BRANCH"
+            BASE_BRANCH="main"
 
-            # Checkout the HEAD SHA
-            git fetch origin $HEAD_SHA
-            git checkout $HEAD_SHA
+            #We are already in the Head of the PR after checkout
```
Comment (geoknee) [inline .circleci/main_config.yml:274]:
```suggestion
            # We are already in the Head of the PR after checkout
```

**Review by geoknee — APPROVED** [review]

**Pull request merged**